hedera-node/src/main/java/com/hedera/services/ledger/TransferLogic.java

```diff
@@ -142,13 +159,20 @@ private void payFunding(final long autoCreationFee) {
 		sideEffectsTracker.trackHbarChange(funding, autoCreationFee);
 	}
 
-	private void adjustHbarUnchecked(final List<BalanceChange> changes) {
+	private void adjustBalancesAndAllowances(final List<BalanceChange> changes) {
 		for (var change : changes) {
+			final var accountId = change.accountId();
 			if (change.isForHbar()) {
-				final var accountId = change.accountId();
 				final var newBalance = change.getNewBalance();
 				accountsLedger.set(accountId, BALANCE, newBalance);
-				sideEffectsTracker.trackHbarChange(accountId, change.units());
+				sideEffectsTracker.trackHbarChange(accountId, change.getAggregatedUnits());
+				if (change.isApprovedAllowance() && change.getAggregatedUnits() < 0) {
```

Comment:
⚠️  Both here and on L172 below, should we use `change.getAllowanceUnits()` instead of `change.getAggregatedUnits()`?

Otherwise, if the txn included `(X, -100, is_approval=true)` **and** `(X, +200, is_approval=false)`, we would not decrease the payer allowance by `100`.

Reply:
ahh yes, `change.isApprovedAllowance()` is enough here.